modMcf/src/org/aion/mcf/config/CfgGuiLauncher.java

```diff
@@ -0,0 +1,125 @@
+package org.aion.mcf.config;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+/**
+ * Configuration for Aion kernel launcher within modGui.
+ *
+ * Represents the <code>launcher</code> subsection within the </code><code>gui</code> section
+ * of Aion kernel config.
+ */
+public class CfgGuiLauncher {
+    private boolean autodetectJavaRuntime;
+    private String javaHome;
+    private String aionSh;
+    private String workingDir;
+    private boolean keepKernelOnExit;
+
+    private String kernelPidFile;
+
+    /**
+     * Instance of a configuration that uses autodetection.  Provided here for convenience.
+     */
+    public static final CfgGuiLauncher AUTODETECTING_CONFIG = new CfgGuiLauncher();
+
+    static {
+        AUTODETECTING_CONFIG.setAutodetectJavaRuntime(true);
+    }
+
+    /** Populate this object from XML data */
+    public void fromXML(final XMLStreamReader sr) throws XMLStreamException {
+        loop:
+        while (sr.hasNext()) {
+            int eventType = sr.next();
+            switch (eventType) {
+                case XMLStreamReader.START_ELEMENT:
+                    String elementName = sr.getLocalName().toLowerCase();
+                    switch (elementName) {
+                        case "autodetect":
+                            this.autodetectJavaRuntime = Boolean.parseBoolean(Cfg.readValue(sr));
+                            break;
+                        case "java-home":
+                            this.javaHome = Cfg.readValue(sr);
+                            break;
+                        case "aion-sh":
+                            this.aionSh = Cfg.readValue(sr);
+                        case "working-dir":
+                            this.workingDir = Cfg.readValue(sr);
+                            break;
+                        case "keep-kernel-on-exit":
+                            this.keepKernelOnExit = Boolean.parseBoolean(Cfg.readValue(sr));
+                            break;
+                        default:
+                            break;
+                    }
+                    break;
+                case XMLStreamReader.END_ELEMENT:
+                    break loop;
+            }
+        }
+    }
+
```

Comment:
Do you need some settings (or default settings) been wrote into the config.xml? if the answer is yes, should update the config.xml in the modBoot/resource.

Reply:
Ok, will update